src/map/script.cpp

```diff
@@ -17571,12 +17571,12 @@ BUILDIN_FUNC(sscanf){
 			if(ref_str==nullptr){
 				CREATE(ref_str, char, strlen(str)+1);
 			}
-			if(sscanf(str, buf, ref_str)==0){
+			if (auto ret = sscanf(str, buf, ref_str); ret == 0 || ret == EOF) {
 				break;
 			}
 			set_reg_str( st, sd, reference_uid( reference_getid( data ), reference_getindex( data ) ), buf_p, ref_str, reference_getref( data ) );
 		} else {  // Number
-			if(sscanf(str, buf, &ref_int)==0){
+			if (auto ret = sscanf(str, buf, &ref_int); ret == 0 || ret == EOF) {
```

Comment:
```suggestion
			if (sscanf(str, buf, &ref_int) != 1) {
```
Why do we check EOF here?
Can we add a warning/error message here?